cmd/prometheus/config.go

```diff
@@ -52,15 +51,37 @@ var cfg = struct {
 	notifierTimeout    time.Duration
 	queryEngine        promql.EngineOptions
 	web                web.Options
-	remote             remote.Options
 
 	alertmanagerURLs stringset
 	prometheusURL    string
-	influxdbURL      string
 }{
 	alertmanagerURLs: stringset{},
 }
 
+// Value type for flags that are now unused, but which are kept around to
+// fulfill 1.0 stability guarantees.
+type unusedFlag struct {
+	name  string
+	value string
+	help  string
+}
+
+func (f *unusedFlag) Set(v string) error {
+	f.value = v
+	log.Errorf("ERROR: flag %q is unused, but set to %q! See the flag's help message: %s", f.name, f.value, f.help)
```

Comment:
I thought "doppelt hält besser" ;)

Hm, I was also split on whether it should be a warning or error. But then I convinced myself that it's really a bad error, as Prometheus will not be behaving as you expect it to, and you will lose data. Wouldn't you agree?